### DISCUSSION TITLE
Does this type of nested parameterisation exist?

### DISCUSSION THREAD

**Discussion by sidekick-eimantas** [pytest-dev/pytest#11280] [Q&A] [answered]
I have a number of openapi.yml schemas and want to validate standardization conformance.
For each path in each schema, I want to invoke a test case.
The simplest solution I can think of is to build a dict of schemas+path-name and the path objects.
I'm wondering if pytest allows setting up parameterization such that, say, one parameterized fixture yields all of the schemas, then another parameterized fixture, for each of those schemas, yields all of the paths?

Essentially something like this (which is ofcourse invalid python):
```py
@pytest.fixture(scope="module", params=api.active_openapi_apis(), ids=[i.full_name for i in api.active_openapi_apis()])
def openapi_api(request) -> api.OpenAPI:
    return request.param

@pytest.fixture(scope="module", params=[path for api in openapi_api for path in api.paths], ids=[path.name for api in openapi_api for path in api.paths])
def path(request) -> str:
    return request.param

def test_path(path: str) -> None:
    assert path.startswith("/")
```

**Comment by RonnyPfannschmidt** [comment, marked as answer]
covarient parameterization currently is not implemented (where parametersets depend on the value of other parametersets

**Comment by The-Compiler** [comment]
Is there a reason you need to use two separate fixtures like this? You could do this if you figure out the combinations in advance, and then either feed it into one fixture as `params`, or used a `pytest_generate_tests` hook ([docs](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html#pytest-generate-tests)).
